aptos-move/framework/src/module_metadata.rs

```diff
@@ -146,8 +146,32 @@ pub fn get_vm_metadata_v0(vm: &MoveVM, module_id: ModuleId) -> Option<RuntimeMod
     }
 }
 
+/// Check if the metadata has unknown key/data types
+pub fn check_metadata_format(module: &CompiledModule) -> Result<(), MalformedError> {
+    if module.metadata.len() > 1 {
+        return Err(MalformedError::MultipleEntries(module.metadata.len() as u64));
```

Comment:
This is an unnecessary restriction. There can be multiple entries in the metadata section. That the keys use `ATPOS_*` naming is perhaps a bit misleading. This is basically a table where you'd only deserialize what is needed from it. For example, we soon may add another key `MOVE_SPECS` or `READ_WRITE_SETS`. Those keys represent other part of metadata.

In the original design, even other tools could have attached metadata, and its probably good for tightening semantics to not do this any longer. We still need multiple keys.

Reply:
Also to be clear, there was never any problem with making this table extensible and having arbitrary, unknown key/value pairs. The problem was a versioning problem, where an older chain allowed a newer APTOS metadata key.  This is a problem is specific to our handling of the metadata keys, and not to the mechanism.

Reply:
Completely arbitrary is dangerous, it means we need to pre-register a series of keys so that we don't end up with a conflict wherein a creative individual breaks future schemes requiring both publish and run time evaluation of metadata.

Do we have a clear definition of what is safely arbitrary?

Reply:
I think you'll need to evaluate this against historical data and verify it passes or does not.

I might suggest that first you make a build that prints out every unknown metadata type from mainnet and testnet. From there, we determine how to structure this and maybe define "safely arbitrary" if we have to.

Reply:
What Wolfgang said makes sense - we should enforce that they're are no unknown keys but shouldn't restrict the number of entries as it'd make it hard to add new entries in the future (multiple places to gate)